lib/common/lifecycle/LifecycleWatcher.dart

```diff
@@ -0,0 +1,103 @@
+import 'dart:ui';
+
+import 'package:flutter/widgets.dart';
+
+/// A mixin that provides lifecycle watching capabilities to a [StatefulWidget]'s State.
+/// It implements [WidgetsBindingObserver] to observe lifecycle changes and provides
+/// hooks for these changes. This allows for a clean and organized way to respond to
+/// lifecycle events without cluttering the main State class.
+mixin LifecycleWatcher<T extends StatefulWidget> on State<T>
+    implements WidgetsBindingObserver {
+  /// Registers the observer on initState.
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addObserver(this);
+  }
+
+  /// Removes the observer on dispose to prevent memory leaks.
+  @override
+  void dispose() {
+    WidgetsBinding.instance.removeObserver(this);
+    super.dispose();
+  }
+
+  /// Responds to lifecycle changes by calling respective methods for each state.
+  @override
+  void didChangeAppLifecycleState(AppLifecycleState state) {
+    switch (state) {
+      case AppLifecycleState.resumed:
+        onResume();
+        break;
+      case AppLifecycleState.inactive:
+        onInactive();
+        break;
+      case AppLifecycleState.paused:
+        onPause();
+        break;
+      case AppLifecycleState.detached:
+        onDetach();
+        break;
+      case AppLifecycleState.hidden:
+        onHidden();
```

Comment:
Nit: Mozna `onHide`, aby to vic sedelo s `resumed` -> `onResume`, `paused` -> `onPause`?

Reply:
Jasně, to dává smysl, opraveno 👍